torch/lib/THPP/tensors/generic/THTensor.cpp

```diff
@@ -113,7 +122,7 @@ auto THTensor<real>::resize(THLongStorage *size,
 
 template<>
 auto THTensor<real>::resizeAs(const Tensor& src) -> THTensor& {
-  THTensor_(resizeAs)(tensor, dynamic_cast<const THTensor<real>&>(src).tensor);
```

Comment:
Nice! Then we can remove `non_const_cast*` altogether, as we never need to drop the const qualifier.